frameworks/hdfs/docs/security.md

```diff
@@ -0,0 +1,145 @@
+- Sufficient access to the KDC to retrieve a keytab for the generated principals
+- [The DC/OS Enterprise CLI](https://docs.mesosphere.com/latest/cli/enterprise-cli/#installing-the-dcos-enterprise-cli)
+- DC/OS Superuser permissions
+
+#### Configure Kerberos Authentication
+
+#### Create principals
+
+The DC/OS Apache HDFS service requires Kerberos principals for each node to be deployed. The overall topology of the HDFS service is:
+- 3 journal nodes
+- 2 name nodes (with ZKFC)
+- A configurable number of data nodes
+
+As such the required Kerberos principals will have the form:
+```
+<service primary>/name-0-node.<service subdomain>.autoip.dcos.thisdcos.directory@<service realm>
+<service primary>/name-0-zkfc.<service subdomain>.autoip.dcos.thisdcos.directory@<service realm>
+<service primary>/name-1-node.<service subdomain>.autoip.dcos.thisdcos.directory@<service realm>
+<service primary>/name-1-zkfc.<service subdomain>.autoip.dcos.thisdcos.directory@<service realm>
+
+<service primary>/journal-0-node.<service subdomain>.autoip.dcos.thisdcos.directory@<service realm>
+<service primary>/journal-1-node.<service subdomain>.autoip.dcos.thisdcos.directory@<service realm>
+<service primary>/journal-2-node.<service subdomain>.autoip.dcos.thisdcos.directory@<service realm>
+
+<service primary>/data-<data-index>-node.<service subdomain>.autoip.dcos.thisdcos.directory@<service realm>
+
+```
+with:
+- `service primary = service.security.kerberos.primary`
+- `data index = 0 up to data_node.brokers.count - 1`
+- `service subdomain = service.name with all `/`'s removed`
+- `service realm = service.security.kerberos.realm`
+
+For example, if installing with these options:
+```json
+{
+    "service": {
+        "name": "a/good/example",
+        "kerberos": {
+            "primary": "example",
+            "realm": "EXAMPLE"
+        }
+    },
+    "data_node": {
+        "count": 3
+    }
+}
+```
+then the principals to create would be:
+```
+example/name-0-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+example/name-0-zkfc.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+example/name-1-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+example/name-1-zkfc.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+
+example/journal-0-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+example/journal-1-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+example/journal-2-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+
+example/data-0-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+example/data-1-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+example/data-2-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+```
+
+{% include services/security-configure-transport-encryption.md
+    techName="Apache HDFS" %}
+
+#### Install the Service
+
+Install the DC/OS Apache HDFS service with the following options in addition to your own:
+```json
+{
+    "service": {
+        "kerberos": {
```

Comment:
This should be a `service.security.kerberos` section.

Reply:
@benclarkwood might have missed this?

Reply:
Ping?